src/draw/AutomaticLayout.js

```diff
@@ -0,0 +1,406 @@
+ * @typedef {object} NodeData
+ * @property {NodeData|null} parent - The node's parent
+ * @property {Component|null} raw The underlying component
+ * @property {NodeData[]} children This node's children
+ * @property {number} depth distance from the root (root's depth is 0)
+ */
+
+/**
+ * Parameters for the ELK automatic layout system.
+ * To be used within {@link AutomaticLayout}.
+ * @typedef {object} ElkParams
+ * @property {boolean} interactive whether we use the interactive mode
+ * @property {boolean} separateConnectedComponents see ELK documentation
+ * @property {string} cycleBreaking see ELK documentation
+ * @property {string} layering see ELK documentation
+ * @property {string} crossingMinimization see ELK documentation
+ * @property {string} nodePlacement see ELK documentation
+ * @property {string} interactiveReferencePoint see ELK documentation
+ * @property {string} direction see ELK documentation
+ */
+
+// eslint-disable-next-line jsdoc/require-property
+/**
+ * An ELK port.
+ * (external)
+ * @typedef {object} ElkPort
+ * @see elk-api.d.ts
+ */
+
+// eslint-disable-next-line jsdoc/require-property
+/**
+ * An ELK edge.
+ * (external)
+ * @typedef {object} ElkExtendedEdge
+ * @see elk-api.d.ts
+ */
+
+/**
+ * An ELK node.
+ * (external)
+ * @typedef {object} ElkNode
+ * @property {string} id id
+ * @property {ElkNode[]} [children] children
+ * @property {ElkPort[]} [ports] ports
+ * @property {ElkExtendedEdge[]} [edges] edges
+ * @see elk-api.d.ts
+ * @example
+ *
+ * // from Typescript
+ * export interface ElkNode extends ElkShape {
+ *     id: string
+ *     children?: ElkNode[]
+ *     ports?: ElkPort[]
+ *     edges?: ElkExtendedEdge[]
+ * }
+ */
+
+/**
+ * Various tools to arrange the graph view (semi-)automatically
+ *
+ * Uses among other things Eclipse Layout Kernel (aka. ELK)
+ *
+ * ELK's documentation is available at:
+ * https://eclipse.dev/elk/
+ *
+ * What we use here elkjs, which is a transpiled version
+ * from Java to Javascript :
+ * https://github.com/kieler/elkjs
+ *
+ */
+
+// TODO : all methods could be static ; do we need a class or a module is enough ?
+export class AutomaticLayout {
+  /**
+   * ELK import
+   * @type {ELK}
+   * @private
+   * @constant
+   */
+  static __elk = new ELK();
+
+  /**
+   * Return components and links associated with a given plugin.
+   * @param {DefaultData} pluginData a leto-modelizer plugin
+   * @returns {{components:Component[], links: ComponentLink[]}} the components and links
+   * @public
+   */
+  static getComponentsAndLinksFromPlugin(pluginData) {
+    return {
+      components: pluginData.components,
+      links: pluginData.getLinks(),
+    };
+  }
+
+  /**
+   * Find the id a node's parent if it has one.
+   * @param {NodeData} node a component data
+   * @returns {string|null} node's parent id or null
+   * @private
+   */
+  __findParentId(node) {
+    // eslint-disable-next-line no-restricted-syntax
```

Comment:
remove this eslint-disable and use Array.find

Reply:
this method has been deleted